cmd/influxd/upgrade/setup.go

```diff
@@ -25,81 +22,35 @@ func setupAdmin(ctx context.Context, v2 *influxDBv2, req *influxdb.OnboardingReq
 	return res, nil
 }
 
-func onboardingRequest(ui *input.UI, options *options) (*influxdb.OnboardingRequest, error) {
-	if (options.force || len(options.target.password) > 0) && len(options.target.password) < internal.MinPasswordLen {
-		return nil, internal.ErrPasswordIsTooShort
-	}
-
-	req := &influxdb.OnboardingRequest{
-		User:                   options.target.userName,
-		Password:               options.target.password,
-		Org:                    options.target.orgName,
-		Bucket:                 options.target.bucket,
-		RetentionPeriodSeconds: influxdb.InfiniteRetention,
-		Token:                  options.target.token,
-	}
-
-	if options.target.retention != "" {
-		dur, err := internal.RawDurationToTimeDuration(options.target.retention)
-		if err != nil {
-			return nil, err
-		}
-		secs, nanos := math.Modf(dur.Seconds())
-		if nanos > 0 {
-			return nil, fmt.Errorf("retention policy %q is too precise, must be divisible by 1s", options.target.retention)
-		}
-		req.RetentionPeriodSeconds = int64(secs)
-	}
-
-	if options.force {
-		return req, nil
-	}
-
-	fmt.Fprintln(ui.Writer, string(internal.PromptWithColor("Welcome to InfluxDB 2.0!", internal.ColorYellow)))
-	if req.User == "" {
-		req.User = internal.GetInput(ui, "Please type your primary username", "")
-	}
-	if req.Password == "" {
-		req.Password = internal.GetPassword(ui, false)
+func onboardingRequest(cli clients.CLI, options *options) (*influxdb.OnboardingRequest, error) {
+	setupClient := setup.Client{CLI: cli}
+	cliReq, err := setupClient.OnboardingRequest(&setup.Params{
+		Username:  options.target.userName,
+		Password:  options.target.password,
+		AuthToken: options.target.token,
+		Org:       options.target.orgName,
+		Bucket:    options.target.bucket,
+		Retention: options.target.retention,
+		Force:     options.force,
+	})
+	if err != nil {
+		return nil, err
 	}
-	if req.Org == "" {
-		req.Org = internal.GetInput(ui, "Please type your primary organization name", "")
+	req := influxdb.OnboardingRequest{
+		User:   cliReq.Username,
+		Org:    cliReq.Org,
+		Bucket: cliReq.Bucket,
 	}
-	if req.Bucket == "" {
-		req.Bucket = internal.GetInput(ui, "Please type your primary bucket name", "")
+	if cliReq.Password != nil {
```

Comment:
So for `influxdb.OnboardingRequest` we don't distinguish between nil and default-value, but for `api.OnboardingRequest` we do?

Reply:
Yes, a consequence of codegen vs. hand-rolled. The codegen'd models use pointer-fields for every optional field.

Reply:
Talked on Slack: going to try deleting `influxdb.OnboardingRequest` and migrating existing code to use the codegen'd model imported from the CLI

Reply:
After trying, would prefer to leave it as an adapter layer here. Both `influxdb.OnboardingRequest` and `api.OnboardingRequest` embed references to other models from their respective packages; if you change the top-level type, you also have to change how you process all the fields, and it starts to sprawl into many files.